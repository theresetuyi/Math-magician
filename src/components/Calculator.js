import React, { useState } from 'react';
import calculate from '../logic/calculate';

function Calculator() {
  const [state, setState] = useState({
    total: 0,
    next: null,
    operation: null,
  });

  const clickListener = (e) => {
    const result = calculate(state, e.target.value);
    setState(result);
  };

  const { total, next, operation } = state;
  const result = `${total || ''}${operation || ''}${next || ''}`;

  return (
    <div>
      <div className="header">Lets do some Math!</div>
      <form action="#">
        <div className="text">{result || '0'}</div>
        <div className="calc-btn">
          <input type="button" value="AC" onClick={clickListener} />
          <input type="button" value="+/-" onClick={clickListener} />
          <input type="button" value="%" onClick={clickListener} />
          <input type="button" value="รท" onClick={clickListener} className="sign" />
          <input type="button" value="7" onClick={clickListener} />
          <input type="button" value="8" onClick={clickListener} />
          <input type="button" value="9" onClick={clickListener} />
          <input type="button" value="x" onClick={clickListener} className="sign" />
          <input type="button" value="4" onClick={clickListener} />
          <input type="button" value="5" onClick={clickListener} />
          <input type="button" value="6" onClick={clickListener} />
          <input type="button" value="-" onClick={clickListener} className="sign" />
          <input type="button" value="1" onClick={clickListener} />
          <input type="button" value="2" onClick={clickListener} />
          <input type="button" value="3" onClick={clickListener} />
          <input type="button" value="+" onClick={clickListener} className="sign" />
          <input type="button" value="0" onClick={clickListener} className="zero" />
          <input type="button" value="." onClick={clickListener} />
          <input type="button" value="=" onClick={clickListener} className="sign" />
        </div>
      </form>
    </div>
  );
}

export default Calculator;
