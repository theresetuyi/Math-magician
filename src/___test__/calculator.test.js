import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Calculation validation:', () => {
  test('Calculator should render', () => {
    const { container } = render(<Calculator />);
    const calcBox = container.querySelector('.calc-box');

    expect(calcBox).toBeInTheDocument();
  });
});

describe('Creating Snapshot for Calculator', () => {
  it('calculator should crete a snapshot correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
