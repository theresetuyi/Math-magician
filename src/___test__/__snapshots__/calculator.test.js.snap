// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating Snapshot for Calculator calculator should crete a snapshot correctly 1`] = `
<div>
  <div
    className="header"
  >
    Lets do some Math!
  </div>
  <form
    action="#"
    className="calc-box"
  >
    <div
      className="text"
    >
      0
    </div>
    <div
      className="calc-btn"
    >
      <input
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        className="sign"
        onClick={[Function]}
        type="button"
        value="รท"
      />
      <input
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        className="sign"
        onClick={[Function]}
        type="button"
        value="x"
      />
      <input
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        className="sign"
        onClick={[Function]}
        type="button"
        value="-"
      />
      <input
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        className="sign"
        onClick={[Function]}
        type="button"
        value="+"
      />
      <input
        className="zero"
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        className="sign"
        onClick={[Function]}
        type="button"
        value="="
      />
    </div>
  </form>
</div>
`;
